samples/create-symbol-styles-from-web-styles/src/main/java/com/esri/arcgismaps/sample/createsymbolstylesfromwebstyles/components/CreateSymbolStylesFromWebStylesViewModel.kt

```diff
@@ -0,0 +1,184 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.createsymbolstylesfromwebstyles.components
+
+import android.app.Application
+import android.graphics.drawable.BitmapDrawable
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+import com.arcgismaps.Color
+import com.arcgismaps.data.ServiceFeatureTable
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.Viewpoint
+import com.arcgismaps.mapping.layers.FeatureLayer
+import com.arcgismaps.mapping.symbology.SymbolStyle
+import com.arcgismaps.mapping.symbology.UniqueValue
+import com.arcgismaps.mapping.symbology.UniqueValueRenderer
+import com.arcgismaps.toolkit.geoviewcompose.MapViewProxy
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialogViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+class CreateSymbolStylesFromWebStylesViewModel(application: Application) : AndroidViewModel(application) {
+
+    val mapViewProxy = MapViewProxy()
+
+    // Create a unique value renderer
+    private val uniqueValueRenderer = UniqueValueRenderer().apply {
+        // Add the name of a field from the feature layer data that symbols will be mapped to
+        fieldNames.add("cat2")
+    }
+
+    // Create a feature layer from a service
+    val featureLayer =
+        FeatureLayer.createWithFeatureTable(ServiceFeatureTable("http://services.arcgis.com/V6ZHFr6zdgNZuVG0/arcgis/rest/services/LA_County_Points_of_Interest/FeatureServer/0"))
+            .apply {
+                // Set the unique value renderer on the feature layer
+                renderer = uniqueValueRenderer
+            }
+
+    /**
+     * Only scale symbols if map scale greater than or equal to 80,000
+     */
+    fun onMapScaleChanged(scale: Double) {
+        featureLayer.scaleSymbols = scale >= 80000
+    }
+
+    // Create a map with the light gray basemap style
+    val arcGISMap = ArcGISMap(BasemapStyle.ArcGISLightGray).apply {
+        // Set a viewpoint
+        initialViewpoint = Viewpoint(34.28301, -118.44186, 7000.0)
+        // Set a reference scale on the map for controlling symbol size
+        referenceScale = 100000.0
+        // Add the feature layer to the map's operational layers
+        operationalLayers.add(featureLayer)
+    }
+
+    // Create a symbol style from a web style. ArcGIS Online is used as the default portal when null is passed as the
+    // portal parameter
+    private val symbolStyle = SymbolStyle.createWithStyleNameAndPortal("Esri2DPointSymbolsStyle", null)
```

Comment:
I think it would be helpful to have named arguments to indicate the `null` property.